File: 0845-longest-mountain-in-array/0845-longest-mountain-in-array.java
class Solution {
    public int longestMountain(int[] arr) {
        int ans=0,start=0;
        int n=arr.length;
        
        while(start<n)
        {
            int end=start;
            if(end+1<n && arr[end]<arr[end+1])
            {
                while(end+1<n && arr[end]<arr[end+1])
                    end++;
                
                
                if(end+1<n && arr[end]>arr[end+1])
                {
                    while(end+1<n && arr[end]>arr[end+1])
                        end++;
                    
                    
                    ans=Math.max(ans,end-start+1);
                }
            }
            start=Math.max(end,start+1);
        }
        return ans;
        
        
    }
}

File: 0845-longest-mountain-in-array/0845-longest-mountain-in-array.cpp
class Solution {
public:
    int longestMountain(vector<int>& a) {
        
        int ans=0;
        int start=0,end=0;
        int n=a.size();
        while(end<n)
        {
            end=start;
            if(end+1<n && a[end]<a[end+1])
            {
                while(end+1<n && a[end]<a[end+1])
                    end++;
                
                if(end+1<n && a[end]>a[end+1])
                {
                    while(end+1<n && a[end]>a[end+1])
                        end++;
                    
                    ans=max(ans,end-start+1);
                }
            }
            start=max(start+1,end);
        }
        return ans;
    }
};